ClientApp/app/components/user/user-form.component.html

```diff
@@ -1,61 +1,99 @@
 ﻿<div *ngIf="model" style="margin-top:20px">
         <form (ngSubmit)="onUserSubmit($event)" #userForm="ngForm">
-            <div><label>Id: {{model.id}}</label></div>
+            <div *ngIf="!newUser"><label>Id: {{model.id}}</label></div>
+
             <div class="form-group">
                 <label for="name">Name</label>
                 <input type="text"
                        class="form-control"
                        id="name"
-                       required
+                       required minlength="4" maxlength="25"
                        [(ngModel)]="model.name"
-                       name="name">
+                       name="name" #name="ngModel">
+                <div *ngIf="name.errors && (name.dirty || name.touched)"
```

Comment:
Repetitive validations we can move into some common place either like here - http://almerosteyn.com/2016/03/angular2-form-validation-component